Source/monster.cpp

```diff
@@ -2784,15 +2804,10 @@ bool M_DumbWalk(int i, Direction md)
 	return ok;
 }
 
-static Direction turn(Direction direction, bool turnLeft)
-{
-	return turnLeft ? left[direction] : right[direction];
-}
-
-bool M_RoundWalk(int i, Direction direction, int *dir)
+bool M_RoundWalk(int i, Direction direction, RotationDirection *spin)

```

Comment:
The only time this updates `monster[i]->mgoalvar2` is if the direction checks fail. I think it'd be easier to take this argument by value as that makes the conversion easier and lets it get passed straight through to `Rotate()`.

This does mean updating `_mgoalvar2` explicitly as a side effect at the end of the function, ideally that side-effect gets removed in a later refactor.

Reply:
Are you suggesting to take `MonsterStruct &` instead of `RotationDirection *` as the second argument?

Reply:
RotationDirection by value as the third argument unless you wanted to unpick the use of the monster struct and array index further down the call stack :)
```cpp
bool M_RoundWalk(int i, Direction direction, RotationDirection spin) {
    // existing logic updated to not dereference spin
    monster[i]._mgoalvar2 = monster[i].mgoalvar2 == RotationDirection::RotateRight ? RotationDirection::RotateLeft : RotationDirection::RotateRight;
}
```

(That said I think only DirOk uses the `i` value in a way that makes refactoring slightly complicated.)

Reply:
Can this be further simplified?
The `spin` parameter is always read from and always written to `_mgoalvar2` or?
```cpp
bool RoundWalk(int i, Direction direction)
{
   RotationDirection spin = static_cast<RotationDirection>(monster[i]._mgoalvar2)
   // existing logic
   monster[i]._mgoalvar2 = static_cast<int>(spin);
}
```